lib/job/compute-next-run-at.ts

```diff
@@ -34,40 +34,68 @@ export const computeNextRunAt = function(this: Job) {
    */
   const computeFromInterval = () => {
     debug('[%s:%s] computing next run via interval [%s]', this.attrs.name, this.attrs._id, interval);
-    let lastRun = this.attrs.lastRunAt || new Date();
-    lastRun = dateForTimezone(lastRun);
-    const cronOptions: any = { currentDate: lastRun.toDate() };
+    const dateNow = new Date();
+    let lastRun: Date = this.attrs.lastRunAt || dateNow;
+    let { startDate, endDate, skipDays } = this.attrs;
+    lastRun = dateForTimezone(lastRun).toDate();
+    const cronOptions: any = { currentDate: lastRun };
     if (timezone) {
       cronOptions.tz = timezone;
     }
 
     try {
       let cronTime = parser.parseExpression(interval, cronOptions);
-      let nextDate = cronTime.next().toDate();
-      if (nextDate.valueOf() === lastRun.valueOf() || nextDate.valueOf() <= previousNextRunAt.valueOf()) {
+      let nextDate: Date | null = cronTime.next().toDate();
+      if (nextDate.getTime() === lastRun.getTime() || nextDate.getTime() <= previousNextRunAt.getTime()) {
```

Comment:
Consistent type.
Variables used to be mixed between: number | Date | moment.Moment